bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/DPIUtil.java

```diff
@@ -731,4 +732,31 @@ public ImageData getImageData(int zoom) {
 		return DPIUtil.scaleImageData(device, imageData, zoom, currentZoom);
 	}
 }
+
+@SuppressWarnings("deprecation")
+public static ImageFileProvider asImageFileProvider(ImageFileNameProvider provider) {
```

Comment:
Why is this placed in DPIUtil`? I don't see any DPI-relation here. Shouldn't it be at some other place for some common Image-related implementations?

Reply:
It's not related at all. I just didn't found a shared/common internal Util class that seems suitable, but I agree that it should be in something Image related.
I have now moved it as package-private method to `ImageData`, but `ImageLoader` would also be possible. I found didn't find one much more suitable than the other. What's your preference or do you have a much better other place?

Reply:
Probably there is no really good place as the best would obviously be inside `Image`, but that class is not common so code would need to be replicated. Adding a utility class only for that method also seems to be a bit inappropriate even though it would probably be the "cleanest" we could do. So maybe `ImageData` is probably the least bad place :-)

One minor addition: with the recent rename of `ImageFileProvider` to `ImagePathProvider` this method should be renamed to `asImagePathProvider` accordingly.

Reply:
> One minor addition: with the recent rename of `ImageFileProvider` to `ImagePathProvider` this method should be renamed to `asImagePathProvider` accordingly.

That's right. Done, thank you.